src/map/skill.c

```diff
@@ -369,6 +369,40 @@ static int skill_get_spiritball(int skill_id, int skill_lv)
 	return skill->dbs->db[idx].spiritball[skill_get_lvl_idx(skill_lv)];
 }
 
+/**
+ * Gets the index of the first required item for a skill at given level.
+ *
+ * @param skill_id The skill's ID.
+ * @param skill_lv The skill's level.
+ * @return The required item's index. Defaults to INDEX_NOT_FOUND (-1) in case of error or if no appropriate index was found.
+ *
+ **/
+static int skill_get_item_index(int skill_id, int skill_lv)
+{
+	if (skill_id == 0)
+		return INDEX_NOT_FOUND;
+
+	Assert_retr(INDEX_NOT_FOUND, skill_lv > 0);
+
+	int idx = skill->get_index(skill_id);
+
+	Assert_retr(INDEX_NOT_FOUND, idx != 0);
+
+	int item_index = INDEX_NOT_FOUND;
+
+	for (int i = 0; i < MAX_SKILL_ITEM_REQUIRE; i++) {
+		if (skill->dbs->db[idx].req_items.item[i].id == 0)
+			continue;
+
+		if (skill->dbs->db[idx].req_items.item[i].amount[skill_get_lvl_idx(skill_lv)] != -1) {
```

Comment:
probably call to ``skill_get_lvl_idx`` better put outside of ``for`` and save value to variable. inside ``for`` only read saved variable with idx

Reply:
Yeah, good point. Changed...